pending-documentation.md

```diff
@@ -185,6 +185,99 @@ have any constructors declared.
 syntax Bytes [hook(BYTES.Bytes), token]
 ```
 
+### Converting between `[token]` sorts
+
+You can convert between tokens of one sort via `String`s by defining functions
+implemented by builtin hooks.
+The hook `STRING.token2string` allows conversion of any token to a string:
+
+```k
+syntax String ::= FooToString(Foo)  [function, functional, hook(STRING.token2string)]
+```
+
+Similarly, the hook `STRING.string2Token` allows the inverse:
+
+```k
+  syntax Bar ::= StringToBar(String) [function, functional, hook(STRING.string2token)]
+```
+
+WARNING: This sort of conversion does *NOT* do any sort of parsing or validation.
+Thus, we can create arbitary tokens of any sort:
+
+```
+StringToBar("The sun rises in the west.")
+```
+
+Composing these two functions lets us convert from `Foo` to `Bar`
+
+```k
+  syntax Bar ::= FooToBar(Foo) [function]
+  rule FooToBar(F) => StringToBar(FooToString(F))
+```
+
+### Parsing comments, and the `#Layout` sort
+
+Productions for the `#Layout` sort are used to describe tokens that are
+considered "whitespace". For example, below, we use it to define lines begining
+with `;` (semicolon) as comments.
```

Comment:
Clarify here that they will be thrown away by the scanner, so the parser will not even see these tokens.